tls/s2n_client_hello_request.c

```diff
@@ -41,6 +41,13 @@ S2N_RESULT s2n_client_hello_request_recv(struct s2n_connection *conn)
     RESULT_ENSURE_REF(conn->config);
     RESULT_GUARD(s2n_client_hello_request_validate(conn));
 
+    /* Maintain the old s2n-tls behavior by default.
+     * Traditionally, s2n-tls has just ignored all hello requests.
+     */
+    if (!conn->config->renegotiate_request_cb) {
```

Comment:
So the new behavior is that we send the "no renegotiate" alert when a callback is set but the connection didn't negotiate secure renegotiation?

Reply:
If I understand what you mean by "new behavior", then sort of. The new behavior is that we send a "no_renegotiation" alert unless the callback is set, secure renegotiation was negotiated, and the application accepts the request. That means that the new behavior triggers an alert even when the application doesn't set the callbacks. Because we saw that issue with warning-level alerts, I want to restore the old behavior of doing nothing if the application doesn't set the callbacks.